clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -345,12 +345,12 @@ def DataMemberAttr : CIR_Attr<"DataMember", "data_member",
   let parameters = (ins AttributeSelfTypeParameter<
                             "", "mlir::cir::DataMemberType">:$type,
                         OptionalParameter<
-                            "std::optional<size_t>">:$memberIndex);
+                            "std::optional<unsigned>">:$member_index);
```

Comment:
This change seems reasonable, but it also affects CIRGen. Would you please update the signature of `getDataMemberAttr` in `CIRGenBuilder.h`? It would be even better if you make it a separate patch .

Reply:
Updated.